### ISSUE TITLE
qgis_process tool for using GRASS algorithms

### ISSUE THREAD

**Issue by Dan-Eli** [qgis/QGIS#38271]
When we type the command C:\Program Files\QGIS 3.14\bin>qgis_process-qgis.bat list it lists all Available Algorithms even the one in the GRASS library.

When I tried using qgis_process with GRASS, I had the following problem:

GRASS GIS 7 folder is not configured. Please configure it before running GRASS GIS 7 algorithms.ERROR: GRASS GIS 7 folder is not configured. Please configure it before running GRASS GIS 7 algorithms.

I had to create a new qgis_process-qgis.bat

@echo off
call "%~dp0\o4w_env.bat"
call "%OSGEO4W_ROOT%\apps\grass\grass78\etc\env.bat"
call qt5_env.bat
call py3_env.bat
@echo off
path %
OSGEO4W_ROOT%\apps\qgis\bin;%OSGEO4W_ROOT%\apps\grass\grass78\lib;%OSGEO4W_ROOT%\apps\grass\grass78\bin;%PATH%
set QGIS_PREFIX_PATH=%OSGEO4W_ROOT:\=/%/apps/qgis
set GDAL_FILENAME_IS_UTF8=YES
rem Set VSI cache to be used as buffer, see #6448
set VSI_CACHE=TRUE
set VSI_CACHE_SIZE=1000000
set QT_PLUGIN_PATH=%OSGEO4W_ROOT%\apps\qgis\qtplugins;%OSGEO4W_ROOT%\apps\qt5\plugins
set PYTHONPATH=%OSGEO4W_ROOT%\apps\qgis\python;%PYTHONPATH%
"%OSGEO4W_ROOT%\apps\qgis\bin\qgis_process.exe" %*
Line 3 is new and line 8 is modified

For QGIS interactive mode there are 2 files (qgis.bat and qgis-grass7.bat) that can be used to launch QGIS depending if we want to launch QGIS with or without GRASS. In batch mode with qgis_process it should be the same 2 files: qgis_process-qgis.bat and qgis_process-qgis-grass7.bat depending if we want to launch QGIS with or without GRASS.

The tests were done on Windows using QGIS 3.14.1

**Comment by gioman** [comment]
@Dan-Eli related to https://github.com/qgis/QGIS/pull/37883 ?

**Comment by Dan-Eli** [comment]
@gioman Yes and No.  Yes for the introduction of the issue.  But No when once want to excute GRASS algorithm with the script  `C:\Program Files\QGIS 3.14\bin>qgis_process-qgis.bat` presently in place in QGIS 3.14.1 you cannot.  This is why I had to modify the script has explained in the issue.

I'm also proposing a correction see last paragraph in the issue. To create a new script.

**Comment by agiudiceandrea** [comment]
Hi @Dan-Eli, it could be better to report the issue opening a new ticket on the OSGeo4W issue tracker https://trac.osgeo.org/osgeo4w/

**Comment by jef-n** [comment]
qgis_process-qgis.bat is created by https://github.com/qgis/QGIS/blob/96bdfac88d9d0b91ad40945341b8e094cef716d2/ms-windows/osgeo4w/package.cmd#L266
